contracts/CompoundFCM.sol

```diff
@@ -0,0 +1,337 @@
+      _marginEngine.termStartTimestampWad(),
+      _marginEngine.termEndTimestampWad()
+    ));
+
+    /// @dev variable cashflow form term start timestamp to now
+    int256 variableCashflowWad = PRBMathSD59x18.mul(
+      variableTokenBalanceWad,
+      variableFactorFromTermStartTimestampToNow
+    );
+
+    /// @dev the total cashflows as a sum of fixed and variable cashflows
+    int256 cashflowWad = fixedCashflowWad + variableCashflowWad;
+
+    /// @dev convert back to non-fixed point representation
+    remainingSettlementCashflow = PRBMathSD59x18.toInt(cashflowWad);
+
+  }
+
+  modifier onlyAfterMaturity () {
+    if (_marginEngine.termEndTimestampWad() > Time.blockTimestampScaled()) {
+        revert CannotSettleBeforeMaturity();
+    }
+    _;
+  }
+
+  /// @notice Settle Trader
+  /// @dev This function lets us settle a fully collateralised fixed taker position post term end timestamp of the MarginEngine
+  /// @dev the settlement cashflow is calculated by invoking the calculateSettlementCashflow function of FixedAndVariableMath.sol (based on the fixed and variable token balance)
+  /// @dev if the settlement cashflow of the trader is positive, then the settleTrader() function invokes the transferMarginToFCMTrader function of the MarginEngine which transfers the settlement cashflow the trader in terms of the underlying tokens
+  /// @dev if settlement cashflow of the trader is negative, we need to update trader's margin in terms of scaled yield bearing tokens to account the settlement casflow
+  /// @dev once settlement cashflows are accounted for, we safeTransfer the scaled yield bearing tokens in the margin account of the trader back to their wallet address
+  function settleTrader() external override onlyAfterMaturity returns (int256 traderSettlementCashflow) {
+
+    // todo: recipient as input to the function
+
+    TraderWithYieldBearingAssets.Info storage trader = traders[msg.sender];
+
+    int256 settlementCashflow = FixedAndVariableMath.calculateSettlementCashflow(trader.fixedTokenBalance, trader.variableTokenBalance, _marginEngine.termStartTimestampWad(), _marginEngine.termEndTimestampWad(), _rateOracle.variableFactor(_marginEngine.termStartTimestampWad(), _marginEngine.termEndTimestampWad()));
+    trader.updateBalancesViaDeltas(-trader.fixedTokenBalance, -trader.variableTokenBalance);
+
+    if (settlementCashflow < 0) {
+      uint256 currentExchangeRate = _ctoken.exchangeRateStored();
+      uint256 updatedTraderMarginInScaledYieldBearingTokens = trader.marginInScaledYieldBearingTokens - uint256(-settlementCashflow).wadDiv(currentExchangeRate);
+      trader.updateMarginInScaledYieldBearingTokens(updatedTraderMarginInScaledYieldBearingTokens);
+    }
+
+    // if settlement happens late, additional variable yield beyond maturity will accrue to the trader
+    uint256 traderMarginInYieldBearingTokens = getTraderMarginInYieldBearingTokens(trader.marginInScaledYieldBearingTokens);
```

Comment:
getTraderMarginInYieldBearingTokens should just be replaced by trader.marginInYieldBearingTokens